framework/include/materials/MaterialData.h

```diff
@@ -101,68 +66,95 @@ class MaterialData
   /// material properties for given element (and possible side)
   void swapBack(const Elem & elem, unsigned int side = 0);
 
-  ///@{
   /**
-   *  Methods for retrieving MaterialProperties object.  These functions
-   *  should NEVER be used to modify the sizes of the MaterialProperties
-   *  objects.
+   * @returns The properties for the state \p state (defaults to zero).
+   *
+   * This should NEVER be used to modify the size of these objects.
    */
-  MaterialProperties & props() { return _props; }
-  MaterialProperties & propsOld() { return _props_old; }
-  MaterialProperties & propsOlder() { return _props_older; }
+  ///{
+  const MaterialProperties & props(const unsigned int state = 0) const;
+  MaterialProperties & props(const unsigned int state = 0);
   ///@}
 
+  template <typename T, bool is_ad>
+  bool haveGenericProperty(const std::string & prop_name) const;
+
   /// Returns true if the regular material property exists - defined by any material.
   template <typename T>
-  bool haveProperty(const std::string & prop_name) const;
+  bool haveProperty(const std::string & prop_name) const
+  {
+    return haveGenericProperty<T, false>(prop_name);
+  }
 
   /// Returns true if the AD material property exists - defined by any material.
   template <typename T>
-  bool haveADProperty(const std::string & prop_name) const;
-
-  template <typename T, bool is_ad>
-  bool haveGenericProperty(const std::string & prop_name) const
+  bool haveADProperty(const std::string & prop_name) const
   {
-    if constexpr (is_ad)
-      return haveADProperty<T>(prop_name);
-    else
-      return haveProperty<T>(prop_name);
+    return haveGenericProperty<T, true>(prop_name);
   }
 
   /**
-   * @{ Methods for retieving a MaterialProperty object
+   * Retreives a material property
```

Comment:
```suggestion
   * Retrieves a material property
```

Reply:
...I'm gonna fix this in the next PR. promise

Reply:
I figured; hence the "Approved" rather than "Required changes"